apps/web/src/services/iam/components/UserGroupChannelSetModal.vue

```diff
@@ -54,10 +58,38 @@ const storeState = reactive({
 const state = reactive<ChannelSetModalState>({
     loading: false,
     channelName: '',
-    selectedProtocolData: {},
     scheduleInfo: notificationChannelCreateFormState.scheduleInfo,
 });
 
+const { mutate: userGroupChannelMutate } = useMutation({
+    mutationFn: (params: UserGroupChannelCreateParameters|UserGroupChannelUpdateParameters) => {
+        if (userGroupPageState.modal.title === i18n.t('IAM.USER_GROUP.MODAL.CREATE_CHANNEL.TITLE')) {
+            return userGroupChannelAPI.create(params as UserGroupChannelCreateParameters);
+        }
+        return userGroupChannelAPI.update(params as UserGroupChannelUpdateParameters);
+    },
+    onSuccess: () => {
+        emit('confirm');
+        queryClient.invalidateQueries({ queryKey: userGroupChannelListQueryKey.value });
+        if (userGroupPageState.modal.title === i18n.t('IAM.USER_GROUP.MODAL.CREATE_CHANNEL.TITLE')) {
+            showSuccessMessage('', i18n.t('IAM.USER_GROUP.MODAL.CREATE_CHANNEL.SUCCESS_MESSAGE'));
+        } else {
+            showSuccessMessage('', i18n.t('IAM.USER_GROUP.MODAL.CREATE_CHANNEL.UPDATE_SUCCESS_MESSAGE'));
+        }
```

Comment:
nitpick: I think it is dangerous to compare with i18n value and title state. How about using modal type?

Reply:
In this modal file!